pkg/scheduler/tests/scheduler_reservation_test.go

```diff
@@ -382,6 +383,86 @@ func TestAddNewNode(t *testing.T) {
 	assert.Equal(t, 0, ms.getPartitionReservations()[appID], "reservations not removed from partition")
 }
 
+// simple reservation one app one queue
+func TestUnReservationAndDeletion(t *testing.T) {
```

Comment:
This needs to run multiple times, one time is not enough to reproduce the issue.

Reply:
It crashed for me and the logs show that the ask is removed before the allocation is processed. If that does not happen the allocation will be confirmed. The checks at the end of the test case should not return no allocations on the nodes as they are not in the correct state etc.
That would cause the test case to fail.

Reply:
I usually only get this reproduced after a few runs.  I don't think it crashes every time.

> If that does not happen the allocation will be confirmed. The checks at the end of the test case should not return no allocations on the nodes as they are not in the correct state etc.

If that does not happen, then the allocation proposal was not made because it is already removed. That doesn't cover the case we are expecting it to cover.

Reply:
Running multiple times only lowers the chance of that happening but it still does not guarantee that the events come in correctly. The only way that could be done is to stop part of the handling or manipulating the objects directly.